src/fastcdc.rs

```diff
@@ -0,0 +1,281 @@
+use super::Engine;
+use std::default::Default;
+use std::cmp;
+use std::mem;
+use Gear;
+
+pub struct FastCDC {
+    current_chunk_size: u64,
+    gear: Gear,
+}
+
+impl Default for FastCDC {
+    fn default() -> Self {
+        FastCDC {
+            current_chunk_size: 0,
+            gear: Gear::default(),
+        }
+    }
+}
+
+
+impl Engine for FastCDC {
+    type Digest = u64;
```

Comment:
This really emphasises that `trait Engine` is an incorrect abstraction for CDC. `FastCDC` doesn't *actually* have a Digest, nor can you roll a single byte.

(Similarly, AE and MAXP can't even pretend to have a digest, because they're not even approximately hash-based)